src/worker_loader.js

```diff
@@ -12,70 +12,20 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
- /* globals PDFJS, Util */
 
 'use strict';
 
-// List of shared files to include;
-var sharedFiles = [
-  'shared/global.js',
-  'shared/util.js'
-];
+importScripts('../node_modules/requirejs/require.js');
 
-// List of other files to include;
-var otherFiles = [
-  'core/network.js',
-  'core/arithmetic_decoder.js',
-  'core/charsets.js',
-  'core/glyphlist.js',
-  'core/jpg.js',
-  'core/metrics.js',
-  'core/bidi.js',
-  'core/chunked_stream.js',
-  'core/jbig2.js',
-  'core/jpx.js',
-  'core/murmurhash3.js',
-  'core/primitives.js',
-  'core/stream.js',
-  'core/crypto.js',
-  'core/font_renderer.js',
-  'core/parser.js',
-  'core/cmap.js',
-  'core/obj.js',
-  'core/ps_parser.js',
-  'core/fonts.js',
-  'core/function.js',
-  'core/colorspace.js',
-  'core/image.js',
-  'core/pattern.js',
-  'core/evaluator.js',
-  'core/annotation.js',
-  'core/document.js',
-  'core/pdf_manager.js',
-  'core/worker.js'
-];
-
-function loadInOrder(index, path, files) {
-  if (index >= files.length) {
-    PDFJS.fakeWorkerFilesLoadedCapability.resolve();
-    return;
-  }
-  PDFJS.Util.loadScript(path + files[index],
-                  loadInOrder.bind(null, ++index, path, files));
+if (typeof Promise !== 'undefined' && Promise.resolve) {
+  // If possible, using Promises queue instead of RequireJS setTimeout.
+  require.s.contexts._.nextTick = (function (promise) {
+    return promise.then.bind(promise);
+  })(Promise.resolve(undefined));
```

Comment:
@Rob--W not sure if it will be an interest to you, but on Chrome, if I remove this hack, testing will start failing: requirejs somewhat stops resolving and worker code is not loaded. I rigged requirejs code with tracing and that shows that it stops issuing importScripts and/or execution of the requirejs logic stops.